code/drasil-lang/Language/Drasil/Expr/Math.hs

```diff
@@ -130,8 +130,8 @@ prodAll v = Operator MulRe (AllDD v Discrete)
 -- TODO: Above only does for Reals
 
 -- | Smart constructor for 'real interval' membership.
-realInterval :: HasUID c => c -> RealInterval Expr Expr -> Expr
-realInterval c = RealI (c ^. uid)
+realInterval :: (HasSymbol c, HasUID c) => c -> RealInterval Expr Expr -> Expr
+realInterval c = RealI (sy c)
```

Comment:
So this is a bit odd for the user: you can put in anything, as long as it has a symbol, and that's what's used, everything else is forgotten. It's not wrong, but I'm not 100% sure that's what we want?

Reply:
From how we use the UID of the `RealI`, I didn't think this makes any large difference, it would just be bringing the computation of `sy (c ^. uid)` up here instead of later down in the code where it would render the symbol.

Sorry, I might be missing a bit of information here actually. Could you please explain what we want from `RealI`? I thought it was just a way of encoding `a < b`, `a < b < c`, and related information so that we can render it properly and display it nicely.

Reply:
The information it is meant to capture is that a particular quantity (which is crucially not a computed quantity, but a named quantity) belongs to a certain interval. That's what needs to be in `Expr`.  In `DisplayExpr`, we can be sloppier, as we won't try to reason about the meaning of it.

Reply:
I never knew that we had a logistical restriction on the UIDs allowed here. I think I fully understand now. Thank you!